src/pages/List.js

```diff
@@ -2,17 +2,36 @@ import React, { useContext } from 'react';
 import Table from '../components/Table';
 import './List.css';
 import { LocationsContext } from '../context/LocationsContext';
+import getDistanceFromLatLonInKm from '../lib/haversineFunction';
 
 export default function List() {
-  const locationsContext = useContext(LocationsContext);
-  const locations = locationsContext.locations.sort((a, b) => {
-    return a.coordinates[0].dist - b.coordinates[0].dist;
+  const { locations, coordinates } = useContext(LocationsContext);
+
+  const sortedLocations = locations.sort((a, b) => {
+    if (a.coordinates[0].dist) {
+      return a.coordinates[0].dist - b.coordinates[0].dist;
+    } else {
+      const c = getDistanceFromLatLonInKm(
```

Comment:
If we had more time, I’d suggest investigating whether we could find a way to leverage the calculated distances we already have for the locations in order not to have to do that work again here. Nice job just getting it working, though!

Reply:
definitely! this was a fix of the bug when we noticed the table wasn't sorting by distance. having these calculations in context and/or as state and then accessing the numbers in Table and InfoWindow would be ideal! Will work on this upcoming week!